src/02-room-details.js

```diff
@@ -49,7 +136,217 @@ function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {}
       "Kit Hopkins Education Wing"
     ]
  */
-function getConnectedRoomNamesById(rooms, id) {}
+
+//getting a string of all the rooms that a room is connected to
+//the room id has to exist in the connects to array in order for you to find the roomId, once you find it, add at the room name to the string
+//loop through the rooms
+//should there be a helper function that will say if this is true(including the id value), will just say push the value or nah
+//if the room id is not right then the error message is `Room with ID of 'incorrect-id' could not be found.
+
+function getConnectedRoomNamesById(rooms, id) {
+  //create an empty array so that if there are values dump the value in there
+  let listOfConnectingRooms = [];
+  // loop through each room on the list roons
+  for (let i = 0; i < rooms.length; i++) {
+    let eachRoom = rooms[i];
+    //in each room loop through the connects to array
+    for (const connectingRoom of eachRoom.connectsTo) {
+      //check if the connect to array has the given id
+      if (connectingRoom.includes(id)) {
+        //if it does then push the value into the given array
+        listOfConnectingRooms.push(eachRoom.name)
+      }
+      if (connectingRoom.includes('incorrect-id')) {
+        return `Room with ID of 'incorrect-id' could not be found.`
```

Comment:
This unfortunately counts as hard code. If the user were to have used an id of 'fakeID234' we would still want them to receive the error message stating `Room with ID of ''fakeID234' could not be found.`

Reply:
One possibility is creating an array and as you iterate through the rooms array you can push the ids there. Then you can add a conditional, where you deem appropriate, that if the connected room id doesnt exist in that array you can return the appropriate error.